common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -829,8 +830,8 @@ public void onData(String streamId, ByteBuffer buf) throws IOException {
       // memory, while still providing the necessary guarantee.
       synchronized (partitionInfo) {
         AppShuffleMergePartitionsInfo info = appShuffleInfo.shuffles.get(partitionInfo.shuffleId);
-        if (isStale(info, partitionInfo.shuffleMergeId) ||
-            isTooLate(info, partitionInfo.reduceId)) {
+        if (isTooLate(info, partitionInfo.reduceId) ||
+            isStale(info, partitionInfo.shuffleMergeId)) {
```

Comment:
Now that you have fixed the issue, checking `isStale` first is better: it is cheaper check to do first.

Reply:
reverted change